### PR TITLE
Refresh RDS IAM token

### PR DESCRIPTION
**Pull request by mr337**
## Description

This PR adds a goroutine to update the IAM token used for RDS authentication. The token is immediately generate to not delay server start and refreshed every 10 minutes. The token updates the database connection details globally used so as connections are created in the pool the latest token is used.

This change is backwards compatible with plain text password authentication.

Semi related (not required PR): https://github.com/transcom/mymove/pull/2620

## Reviewer Notes

Few things to check:
* Use of goroutine applicable
* Is the error handling sufficient?
* Is there proper logging to debug this if it fails?
* If the IAM token refresh fails the app fill block indefinitely. Is this proper behavior or should we have a max timeout that the server exits after X minutes?

## Setup
If using MilMove AWS environment refer to [IAM Doc](https://github.com/transcom/ppp-infra/blob/master/docs/runbook/0009-accessing-the-db-with-iam.md) on how to connect.

Testing IAM auth
```sh
 ./bin/milmove serve --db-iam --db-iam-role DB_ARN  \
 --db-region us-east-2 \
--db-host rdstest.chsgqg6ccrq7.us-east-2.rds.amazonaws.com  \
--db-ssl-mode verify-full --db-ssl-root-cert  PATH/TO/CERT --db-user db_user
```

Browse for longer than 15 minutes as the original IAM token will have expired.

Every 10 minutes the server will refresh the token before the 15m window:
```
2019-09-23T19:22:17.082Z        INFO    iampostgres/iampostgres.go:68   Using IAM Authentication
2019-09-23T19:22:17.327Z        INFO    iampostgres/iampostgres.go:75   Successfully generated new IAM token
2019-09-23T19:32:17.087Z        INFO    iampostgres/iampostgres.go:68   Using IAM Authentication
2019-09-23T19:32:17.088Z        INFO    iampostgres/iampostgres.go:75   Successfully generated new IAM token
```

Testing Plain Text: simply run local dev environment which uses plain text password for database auth

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/tree/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/tree/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/tree/master/docs/database.md#zero-downtime-migrations))
  * [ ] Have been communicated to #dp3-engineering
  * [ ] Secure migrations have been tested using `scripts/run-prod-migrations`
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/168068716) for this change

### CODE REVIEW

File: pkg/iampostgres/iampostgres.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/service/rds/rdsutils"
+	"go.uber.org/zap"
+
+	pg "github.com/lib/pq"
+)
+
+var useIAM = false
+var passHolder = ""
+var currentIamPass = ""
```
Comment (chrisgilmerproj) [inline pkg/iampostgres/iampostgres.go]: Let's make this a Struct at some point.

**Comment by mr337** [comment]
Decided to revert a commit that used the existing pq driver for plain text auth and the custom driver for RDS. This was a work around for the real problem that affects both plain text and IAM auth for the custom driver. CI was telling me something and I ignored it, so now it will always test both ways.

More info here: https://github.com/jmoiron/sqlx/issues/559

File: go.mod [hunk 2]
```diff
@@ -88,3 +88,6 @@ require (
 	gopkg.in/ini.v1 v1.44.0 // indirect
 	gopkg.in/urfave/cli.v1 v1.20.0 // indirect
 )
+
+// mr337/sqlx v1.2.5 is just jmoiron's 1.2.0 with custom driver fixes
+replace github.com/jmoiron/sqlx v1.2.0 => github.com/mr337/sqlx v1.2.5
```
Comment (chrisgilmerproj) [inline go.mod]: I'd like it if we put this repo in the Transcom Org instead of your personal account.

Comment (chrisgilmerproj) [reply go.mod]: Also, how do we follow up on the status of this? Can you link `jmoiron`'s PR requests link in the comment?

File: pkg/cli/dbconn.go [hunk 3]
```diff
@@ -283,6 +268,22 @@ func InitDatabase(v *viper.Viper, creds *credentials.Credentials, logger Logger)
 		Pool:     dbPool,
 		IdlePool: dbIdlePool,
 	}
+
+	if v.GetBool(DbIamFlag) {
+		// Set password holder for IAMPostgres to easily repalce with temp password
```
Comment (chrisgilmerproj) [inline pkg/cli/dbconn.go]:
```suggestion
		// Set a bogus password for IAMPostgres. It will be replaced with an RDS auth token as the password
```

**Comment by mr337** [comment]
Deployed to experimental. Made a move as a service member waiting to be approved. Check for error and seeing errors about personal access code which has been present for the entire month of Sept, looks to not be related to this PR. 

Checking ECS logs token is being refreshed roughly every 10 minutes as expected.
Screen Shot 2019-09-30 at 11 01 25 AM

**Comment by mr337** [comment]
> One last thing I can think of is enabling this for staging: https://github.com/transcom/mymove/blob/master/config/env/staging.env#L23

This is now done!

File: config/env/staging.env [hunk 4]
```diff
@@ -21,5 +21,5 @@ SERVE_API_INTERNAL=true
 SERVE_API_PUBLIC=true
 MUTUAL_TLS_ENABLED=true
 DB_IAM=true
-DB_REGION=us-east-2
+DB_REGION=us-west-2
```
Comment (mr337) [inline config/env/staging.env]: @chrisgilmerproj This doesn't feel right and not for sure if an antipattern. Without this the acceptance test fails with a:
Screen Shot 2019-09-30 at 1 33 01 PM

Is this OK to do?

Comment (chrisgilmerproj) [reply config/env/staging.env]: Yeah, I think you caught a bug. We don't have anything in `us-east-2` only `us-west-2`. Probably that was my bad. Thanks for fixing it.

What's happening is that previously the code implied the DB_REGION but it wasn't explicitly set by the docker task definition. Having it here makes it explicit what we're doing with RDS IAM.

**Pull request merged**